### PR TITLE
[Feature] Support partition prune and predicate push down on delta lake table

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

#11972 

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 108 (04.63%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/delta/ExpressionConverter.java|0|84|00.00%|[50, 54, 55, 56, 59, 60, 62, 67, 68, 69, 70, 71, 74, 75, 76, 78, 79, 80, 81, 82, 85, 90, 91, 92, 94, 95, 96, 98, 104, 105, 106, 108, 109, 110, 111, 113, 115, 117, 119, 121, 123, 125, 127, 133, 134, 135, 137, 138, 139, 140, 141, 142, 143, 145, 146, 147, 148, 150, 155, 156, 159, 160, 162, 165, 167, 169, 171, 173, 178, 182, 185, 186, 187, 188, 191, 193, 198, 199, 201, 202, 203, 205, 211, 216]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|0|19|00.00%|[49, 54, 78, 79, 80, 81, 82, 84, 85, 86, 87, 89, 91, 92, 93, 94, 108, 113, 115]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownMinMaxConjunctsRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergePredicateRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergeLimitDirectRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RemoteScanPartitionPruneRule.java|1|1|100.00%|[]|

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergePredicateRule.java [hunk 1]
```diff
@@ -34,6 +34,8 @@ public class MergePredicateRule extends TransformationRule {
             new MergePredicateRule(OperatorType.LOGICAL_HUDI_SCAN);
     public static final MergePredicateRule ICEBERG_SCAN =
             new MergePredicateRule(OperatorType.LOGICAL_ICEBERG_SCAN);
+    public static final MergePredicateRule DELTALAKE_SCAN =
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergePredicateRule.java:37]: This rule wiil delete later

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergePredicateRule.java]: done

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Review by Youngwb — APPROVED** [review]

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by Youngwb** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 107 (03.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/delta/ExpressionConverter.java|0|84|00.00%|[50, 54, 55, 56, 59, 60, 62, 67, 68, 69, 70, 71, 74, 75, 76, 78, 79, 80, 81, 82, 85, 90, 91, 92, 94, 95, 96, 98, 104, 105, 106, 108, 109, 110, 111, 113, 115, 117, 119, 121, 123, 125, 127, 133, 134, 135, 137, 138, 139, 140, 141, 142, 143, 145, 146, 147, 148, 150, 155, 156, 159, 160, 162, 165, 167, 169, 171, 173, 178, 182, 185, 186, 187, 188, 191, 193, 198, 199, 201, 202, 203, 205, 211, 216]|
|:large_blue_circle:|com/starrocks/planner/DeltaLakeScanNode.java|0|19|00.00%|[49, 54, 78, 79, 80, 81, 82, 84, 85, 86, 87, 89, 91, 92, 93, 94, 108, 113, 115]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownMinMaxConjunctsRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergeLimitDirectRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RemoteScanPartitionPruneRule.java|1|1|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Pull request merged**